enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,466 @@
+// cluster claim selector matches no objects.
+type ClusterClaimSelector struct {
+	// matchClaims is a map of {key,value} pairs. A single {key,value} in the matchClaims
+	// map is equivalent to an element of matchExpressions, whose key field is "key", the
+	// operator is "In", and the values array contains only "value". The requirements are ANDed.
+	// +optional
+	MatchClaims map[string]string `json:"matchClaims,omitempty"`
+	// matchExpressions is a list of cluster claim selector requirements. The requirements are ANDed.
+	// +optional
+	MatchExpressions []metav1.LabelSelectorRequirement `json:"matchExpressions,omitempty"`
+}
+
+// Affinity is a group of affinity scheduling rules.
+type Affinity struct {
+	// ClusterAntiAffinity describes cluster affinity scheduling rules for the placement.
+	// +optional
+	ClusterAntiAffinity *ClusterAntiAffinity `json:"clusterAntiAffinity,omitempty"`
+}
+
+// ClusterAntiAffinity is a group of cluster affinity scheduling rules.
+type ClusterAntiAffinity struct {
+	// RequiredDuringSchedulingRequiredDuringExecution is a scheduling strategies for
+	// ClusterAntiAffinity. The anti-affinity requirements specified by this field should be met
+	// at both scheduling time (when the placement decisions are made for the first time) and
+	// execution time (After one or multiple placement decisions are made). It means,
+	// 1) Any cluster, which does not meet the requirements, should not be selected;
+	// 2) If a selected cluster ceases to meet the requirements (e.g. due to an update) after the
+	//    placement decisions are made, it will be eventually removed from the placement decisions;
+	// 3) If a cluster starts to meet the requirements after the placement decisions are made, it
+	//    will either be selected or at least has chance to be selected;
+	// When there are multiple elements, the lists of clusters corresponding to each
+	// ClusterAffinityTerm are intersected, i.e. all terms must be satisfied.
+	//
+	// Other scheduling strategies not yet implemented includes
+	// RequiredDuringSchedulingIgnoredDuringExecution and PreferredDuringSchedulingIgnoredDuringExecution
+	// +optional
+	RequiredDuringSchedulingRequiredDuringExecution []ClusterAffinityTerm `json:"requiredDuringSchedulingRequiredDuringExecution,omitempty"`
+}
+
+// The selected clusters could not co-located with each other,
+// where co-located is defined as the value of the label/claim with name <topologyKey> is the same.
+type ClusterAffinityTerm struct {
+	// TopologyKey is either a label key or a cluster claim name of clusters
+	// +required
+	TopologyKey string `json:"topologyKey"`
+
+	// TopologyKeyType indicates the type of TopologyKey. It could be Label or Claim.
+	// +required
+	TopologyKeyType string `json:"topologyKeyType"`
+}
+
+const (
+	TopologyKeyTypeLabel string = "Label"
+	TopologyKeyTypeClaim string = "Claim"
+)
+
+type PlacementStatus struct {
+	// Conditions contains the different condition statuses for this Placement.
```

Comment:
I think we decide to put also number of selected clusters in the status

Reply:
Added.